File: Programmers/Hash/완주하지 못한 선수.js
function solution(participant, completion) {
    let answer = '';
    let m = new Map();
    participant.map((el,idx)=>{
        if(!m.has(el)){
            m.set(el,1);
        }
        else {
            m.set(el,m.get(el)+1);
        }
    })
    
    completion.map((el,idx)=>{
        if(m.has(el)){
            m.set(el,m.get(el)-1);
        }
    })
    
    // m.forEach((value,key)=>{
    //     if(value>=1){
    //         answer += key;
    //     }
    // })
    
    for(const [key ,value] of m){
        if(value >=1){
            answer = key;
        }
    }
    return answer;
}

File: Programmers/Hash/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <map>
#include <iostream>
using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    map <string,int> m;
    for(int i = 0 ; i <participant.size();i++){
        m[participant[i]]++;
    }
    for(int i = 0 ; i <completion.size();i++){
        m[completion[i]]--;
    }
    
    for(auto itr : m){
        // answer.push_back(itr.first);
        if(itr.second > 0 )
            answer = itr.first;
    }
    return answer;
}
